### PR TITLE
Check GoB for mirrored commit before scheduling postsubmit tasks

### PR DESCRIPTION
**Pull request by keyonghan**
This is a rebase of https://github.com/flutter/cocoon/pull/2411 to TOT, and some cleanup.

Fixes: https://github.com/flutter/flutter/issues/62375, https://github.com/flutter/flutter/issues/100911

### CODE REVIEW

**Review by CaseyHillers — APPROVED** [review]
flutter-lgtm

File: app_dart/test/src/datastore/fake_config.dart [hunk 1]
```diff
@@ -267,7 +267,15 @@ class FakeConfig implements Config {
   Future<List<String>> get releaseAccounts async => <String>['dart-flutter-releaser'];
 
   @override
-  Set<gh.RepositorySlug> get supportedRepos => supportedReposValue ?? <gh.RepositorySlug>{Config.flutterSlug};
+  Set<gh.RepositorySlug> get supportedRepos =>
```
Comment (CaseyHillers) [inline app_dart/test/src/datastore/fake_config.dart:270]: Nice work on all the cleanup!

File: app_dart/test/request_handlers/github/webhook_subscription_test.dart [hunk 2]
```diff
@@ -2707,13 +2712,38 @@ void foo() {
...
         merged: true,
+        baseSha: 'sha1', // Found in pre-populated commits in FakeGerritService.
       );
 
       expect(db.values.values.whereType<Commit>().length, 0);
       await tester.post(webhook);
       expect(db.values.values.whereType<Commit>().length, 1);
     });
 
+    test('Fail when pull request is closed and merged, but merged commit is not found on GoB', () async {
```
Comment (ricardoamador) [inline app_dart/test/request_handlers/github/webhook_subscription_test.dart:2723]: Is the base case covered when the commit is found?

**Review by ricardoamador — APPROVED** [review]
LGTM with question

File: app_dart/test/request_handlers/github/webhook_subscription_test.dart (hunk 2 above)
Comment (keyonghan) [reply app_dart/test/request_handlers/github/webhook_subscription_test.dart:2723]: Yeah, please refer to test `Acts on closed, cancels presubmit targets, add pr for postsubmit target create`.

Comment (ricardoamador) [reply app_dart/test/request_handlers/github/webhook_subscription_test.dart:2723]: Thanks!

**Comment by keyonghan** [comment]
This is blocked by https://github.com/flutter/flutter/issues/124873

**Comment by keyonghan** [comment]
Two new changes:
1) `GitPersonInfo` has changed `time` to `date`: https://gerrit-review.googlesource.com/Documentation/rest-api-changes.html#git-person-info. Updated accordingly.
2) Use `Uri` instead of `Uri.https` to generate the url.

**Comment by keyonghan** [comment]
Validated this works. Found the commit and scheduled on the third attempt:
Screenshot 2023-04-20 at 7 36 25 PM




```
> INFO 2023-04-21T01:36:36.189Z Gerrit get-commit url: https://flutter-review.googlesource.com/projects/mirrors%2Fengine/commits/007a2c6889240bdbf5966aced874d7e2b35134cc
> INFO 2023-04-21T01:36:36.738Z Gob commit response for commit 007a2c6889240bdbf5966aced874d7e2b35134cc: )]}' {"commit":"007a2c6889240bdbf5966aced874d7e2b35134cc","parents":[{"commit":"122c3b3820121e60264ea9631dfa77380f709549","subject":"[Windows] Don\u0027t block raster thread until v-blank (#41231)"}],"author":{"name":"Casey Hillers","email":"chillers@google.com","date":"2023-04-21 01:36:06.000000000","tz":-420},"committer":{"name":"GitHub","email":"noreply@github.com","date":"2023-04-21 01:36:06.000000000","tz":-420},"subject":"Revert \"Manual roll skia to d5b4acfb4\" (#41381)","message":"Revert \"Manual roll skia to d5b4acfb4\" (#41381)\n\nReverts flutter/engine#41378\r\n\r\nTests are passing at\r\nhttps://dart.googlesource.com/sdk/+/7247a1b3b7166c828a0c2e00259ec03fd737d455"}
> INFO 2023-04-21T01:36:36.739Z Merged commit was found on GoB mirror. Scheduling postsubmit tasks...
> INFO 2023-04-21T01:36:36.749Z Scheduling 007a2c6889240bdbf5966aced874d7e2b35134cc via GitHub webhook
```

**Pull request merged**